.github/workflows/ci.yaml

```diff
@@ -278,7 +281,8 @@ jobs:
       (github.event_name == 'push') &&
       (github.ref == 'refs/heads/main' ||
        startsWith(github.ref, 'refs/heads/release-') ||
-       startsWith(github.ref, 'refs/tags/v'))
+       startsWith(github.ref, 'refs/tags/v')) &&
+      (github.event.schedule != '0 0 * * *')
```

Comment:
This is kind of fragile - if we change the cron spec to start at 12:30 AM this check will break.

Do we have another way to detect nightly build?

One way can be to set a variable for nightly build (e.g. nightly_build: true) at the start of the build (maybe it is possible as part of the rules?) and check here the flag.